wadl/survey.py

```diff
@@ -157,19 +160,23 @@ def view(self, showGrid=True, save=None):
         else:
             plt.show()
 
-    def plan(self, write=True, showPlot=False):
+    def plan(self, write=True, showPlot=False, relinking=False):
         """ Plan the survey.
 
         Args:
             write (bool): write the paths as csv file and save the plot of
                 the route. default True
             showPlot (bool): show the plot of the routes. default False.
         """
+        # create dict mapping maze to solver
+        self.solvers = dict()
         for task, maze in self.tasks.items():
             self.solver.setup(maze.graph)
             try:
                 solTime = self.solver.solve(routeSet=maze.routeSet)
                 maze.solTime = solTime
+                # store copy of paths
+                self.solvers[maze] = self.solver.copy
```

Comment:
you can just create a new `Solver` instance

Reply:
I'm a little unsure what you mean here. Wouldn't I lose track of the subPaths and other info then?

Reply:
not if you initialize the solver object in the right spot.

`self.solver = LinkSolver()` is being called on the survey init so you can move that to `addTask` so that when a task is added a solver for that task/maze is also created in `self.solvers`